### PR TITLE
 EDU-1650: Adds push lifecycle section and adds push lifecycle diagram

### PR DESCRIPTION
**Pull request by franrob-projects**
This PR:

- Adds a new push notification lifecycle section to the overview page.
- Adds the actual push notification lifecycle diagram.
- Adds an explanation of the push notification diagram.

[EDU-1650](https://ably.atlassian.net/browse/EDU-1650)

[EDU-1650]: https://ably.atlassian.net/browse/EDU-1650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/push/index.textile [hunk 1]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+
+<a href="@content/diagrams/push-lifecycle.png" target="_blank">
+  <img src="@content/diagrams/push-lifecycle.png" style="width: 100%" alt="Push notifications lifecycle">
+</a>
+
+The following explains the push notification lifecycle:
+
+* Publish a push notification:
+** The process begins with your server, which publishes a push notification. This server could be any backend service that needs to send notifications to users' devices.
+** Your server sends the notification data to Ably, a realtime messaging platform.
```
Comment (owenpearson) [inline content/push/index.textile:76]: i think it reads a bit funny briefly explaining what ably is, anyone reading this should already be aware of this so i'd opt to omit this.

```suggestion
** Your server sends the notification data to Ably.
```

File: content/push/index.textile [hunk 2]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+* Handle push notification:
+** After receiving the push notification from your server, Ably processes and packages the notification with any additional data payload. This payload may include data such as message content, metadata, or any other relevant information needed for the notification.
+** Ably then forwards the packaged message to the appropriate push notification service _FCM, APNs, or Web Push_.
```
Comment (owenpearson) [inline content/push/index.textile:79]: web push is not a push notification service, rather a specification which is implemented with different push services depending on the browser.
```suggestion
** Ably then forwards the packaged message to the appropriate push notification service _FCM_, _APNs_, or in the case of Web Push, the browser vendor's notification service.
```

File: content/push/index.textile [hunk 3]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+* Deliver push notification:
+** The selected push notification service _FCM, APNs, or Web Push_ takes responsibility for delivering the notification to the target devices.
```
Comment (owenpearson) [inline content/push/index.textile:81]: see above regarding 'web push' as a push service

File: content/push/index.textile [hunk 4]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+** The push notification finally arrives at the end-use's devices or browsers, displaying the notification on their screen.
```
Comment (owenpearson) [inline content/push/index.textile:82]:
```suggestion
** The push notification finally arrives at the end-user's devices or browsers, displaying the notification on their screen.
```

**Review by owenpearson — CHANGES_REQUESTED** [review]
looks fine overall, some nitpicks:

**Comment by franrob-projects** [comment]
> looks fine overall, some nitpicks:

Hi @owenpearson thanks for the feedback. Please see: 65343c9be2096db9c8d2253e5529cd325b9e6c3b

**Review by owenpearson — APPROVED** [review]
LGTM

**Review by m-hulbert — COMMENTED** [review]
My only comment is that both the text and diagram assume that a publish originates from a server... it may be the case that a client publishes to a push-enabled channel.

File: content/push/index.textile [hunk 5]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+The following diagram shows the push notification lifecycle:
+
+<a href="@content/diagrams/push-lifecycle.png" target="_blank">
+  <img src="@content/diagrams/push-lifecycle.png" style="width: 100%" alt="Push notifications lifecycle">
+</a>
+
+The following explains the push notification lifecycle:
+
+* Publish a push notification:
+** The process begins with your app (client) or server, which publishes a push notification -- this could be any backend service that needs to send notifications to users' devices.
```
Comment (m-hulbert) [inline content/push/index.textile:75]:
```suggestion
** The process begins with your app (client) or server, which publishes a push notification.
```

File: content/push/index.textile [hunk 6]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+** Your app or server sends the notification data to Ably.
+* Handle push notification:
+** After receiving the push notification from your app or server, Ably processes and packages the notification with any additional data payload. This payload may include data such as message content, metadata, or any other relevant information needed for the notification.
```
Comment (m-hulbert) [inline content/push/index.textile:78]:
```suggestion
** After receiving the push notification, Ably processes and packages the notification.
```

File: content/push/index.textile [hunk 7]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+** Ably then forwards the packaged message to the appropriate push notification service _FCM_, _APNs_, or in the case of Web Push, the browser vendor's notification service.
```
Comment (m-hulbert) [inline content/push/index.textile:79]: >browser vendor's notification service

Is this correct?

File: content/push/index.textile [hunk 8]
```diff
@@ -61,6 +61,26 @@ If you publish via channels, devices or browsers must subscribe to those channel
...
+* Deliver push notification:
+** The selected push notification service takes responsibility for delivering the notification to the target devices.
+** The push notification finally arrives at the end-user's devices or browsers, displaying the notification on their screen.
```
Comment (m-hulbert) [inline content/push/index.textile:82]:
```suggestion
** The push notification arrives at the end-user's devices or browsers, displaying the notification on their screen.
```

**Review by m-hulbert — CHANGES_REQUESTED** [review]

**Review by m-hulbert — APPROVED** [review]

**Pull request merged**